_posts/en/newsletters/2019-12-04-newsletter.md

```diff
@@ -0,0 +1,155 @@
+    believes it's not particularly useful compared to just keeping
+    `OP_CODESEPARATOR` in tapscript.
+
+    In a separate schnorr-related topic, ZmnSCPxj wrote a [post][zmn
+    composable musig] about the challenges of safely using the [MuSig][]
+    signature aggregation protocol with sub-groups.  For example,
+    ZmnSCPxj's [nodelets proposal][] would like to allow Alice and Bob
+    to jointly control some funds through a single LN node, using the
+    aggregation of their keys, (A, B).  Their joint node could then open
+    a channel to Charlie's node, using MuSig aggregation there too, ((A,
+    B), C).  However, ZmnSCPxj describes why this might be unsafe given
+    Wagner's algorithm as described in [last week's newsletter][news74
+    taproot updates].  Also described are several alternative schemes
+    that attempt to work around the problem.
+
+- **`OP_CHECKTEMPLATEVERIFY` (CTV):** the successor to
+  `OP_CHECKOUTPUTSHASHVERIFY` (COSHV) described in [Newsletter
+  #48][news48 coshv] and `OP_SECURETHEBAG` mentioned in [Newsletter
+  #49][news49 stb], this new opcode [proposed][ctv post] by Jeremy Rubin
+  would allow a script to ensure its funds could only be spent by a
+  specified set of child transactions.  In addition to the name change,
+  Rubin has also added additional details to the [proposed BIP][bip-ctv]
+  and he plans to hold a review workshop in the first few months of 2020
+  (fill out [this form][ctv workshop] if you're interested in
+  attending).
+
+    On-list, Russell O'Connor [restated][oconnor state variable] a
+    [previous concern][oconnor suggested amendments] of his about CTV
+    pulling data off the stack in an abnormal order for Bitcoin Script.
+    This was added by Rubin in order to prevent the creation of
+    recursive [covenants][topic covenants]---script conditions that
+    apply not just to a finite set of descendant transactions but which
+    will apply to an infinite set of descendant transactions.  For
+    example, a spender could ensure that their coins could only ever
+    again be sent to a set of three pubkeys---any payment to any other
+    pubkey would be forbidden.  O'Connor's concerns seemed to focus on
+    this odd behavior making it harder to model the semantics of Bitcoin
+    Script, something which O'Connor has previously worked on and which
+    is related to his continuing work on the [Simplicity][] scripting
+    language.
+
+    On IRC, Gregory Maxwell and Jeremy Rubin [discussed][irc ctv]
+    several aspects of CTV, especially focusing on making the proposed
+    opcode easier to use with advanced designs without making it harder
+    to use with the simple [congestion controlled transactions][] and
+    [payment pools][] already proposed.  They also discussed whether it
+    was really necessary to prevent people from creating recursive
+    covenants, with possible allusions in the conversation to a [2013
```

Comment:
s/possible allusions/a possible allusion/?

I'm still reading these discussions, but if anyone is interested, IIUC that allusion is here: https://freenode.irclog.whitequark.org/bitcoin-wizards/2019-11-28#25861296.